src/utils/dom-writer.js

```diff
@@ -103,7 +103,7 @@ export class DomWriterStreamer {
     /** @private {boolean} */
     this.eof_ = false;
 
-    /** @private {?Element} */
+    /** @private {?Promise<!Element>} */
```

Comment:
Shouldn't this be something like `{!Promise<?Element>}`? Right now the element is optional but it appears you are making it required. Or are you only resolving the promise if there is an element?

Reply:
I'd expect that body must always be available when the promise resolves.